src/Tizen.NUI/src/public/Images/EncodedImageBuffer.cs

```diff
@@ -0,0 +1,165 @@
+            return ret;
+        }
+
+        /// <summary>
+        /// Dispose
+        /// </summary>
+        /// <param name="type"></param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void Dispose(DisposeTypes type)
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            if (type == DisposeTypes.Explicit)
+            {
+                //Called by User
+                //Release your own managed resources here.
+                //You should release all of your own disposable objects here.
+                mCachedImageUrl?.Dispose();
+            }
+
+            base.Dispose(type);
+        }
+
+        /// This will not be public opened.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void ReleaseSwigCPtr(HandleRef swigCPtr)
+        {
+            Interop.EncodedImageBuffer.DeleteEncodedImageBuffer(swigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Get VectorUnsignedChar from System.IO.Stream.
+        /// This funcion is private (exist only for Constructor.)
+        /// </summary>
+        private static VectorUnsignedChar GetRawBuffrFromStreamHelper(System.IO.Stream stream)
+        {
+            if(stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if(!stream.CanRead)
+            {
+                throw new InvalidOperationException("stream don't support to read");
+            }
+
+            // Copy stream to memoryStream
+            System.IO.MemoryStream memoryStream = new System.IO.MemoryStream();
+            stream.CopyTo(memoryStream);
+            memoryStream.Seek(0, System.IO.SeekOrigin.Begin);
+
+            long streamLength = memoryStream.Length;
+            if(streamLength <= 0)
+            {
+                throw new InvalidOperationException("stream length is <= 0");
+            }
+
+            // Read data from memoryStream
+            byte[] streamData = new byte[streamLength];
+            int readableStreamLength = memoryStream.Read(streamData, 0, (int)streamLength);
+            if(readableStreamLength <= 0)
+            {
+                throw new InvalidOperationException("readable stream length is <= 0");
+            }
+
+            // Allocate buffer that internal DALi engine can read
+            VectorUnsignedChar buffer = new VectorUnsignedChar();
+
+            buffer.Resize((uint)readableStreamLength);
+            var bufferBegin = buffer.Begin();
+            global::System.Runtime.InteropServices.HandleRef bufferRef = SWIGTYPE_p_unsigned_char.getCPtr(bufferBegin);
+
+            // Copy data from memoryStream to buffer
+            System.Runtime.InteropServices.Marshal.Copy(streamData, 0, bufferRef.Handle, readableStreamLength);
```

Comment:
How about use `GetBuffer` instead copy into `streamData`,
https://docs.microsoft.com/ko-kr/dotnet/api/system.io.memorystream.getbuffer?view=net-5.0
```suggestion

            // Allocate buffer that internal DALi engine can read
            VectorUnsignedChar buffer = new VectorUnsignedChar();

            buffer.Resize((uint)streamLength);
            var bufferBegin = buffer.Begin();
            global::System.Runtime.InteropServices.HandleRef bufferRef = SWIGTYPE_p_unsigned_char.getCPtr(bufferBegin);

            // Copy data from memoryStream to buffer
            System.Runtime.InteropServices.Marshal.Copy(memoryStream.GetBuffer(), 0, bufferRef.Handle, streamLength);
```

Reply:
Good points. Fix done.